src/NHibernate/Async/Driver/DriverBase.cs

```diff
@@ -0,0 +1,61 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by AsyncGenerator.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+using System.Linq;
+using NHibernate.Engine;
+using NHibernate.SqlCommand;
+using NHibernate.SqlTypes;
+using NHibernate.Util;
+using Environment = NHibernate.Cfg.Environment;
+
+namespace NHibernate.Driver
+{
+	using System.Threading.Tasks;
+	using System.Threading;
+	public abstract partial class DriverBase : IDriver, ISqlParameterFormatter
+	{
+
+		/// <summary>
+		/// Begin an ADO <see cref="DbTransaction" />.
+		/// </summary>
+		/// <param name="isolationLevel">The isolation level requested for the transaction.</param>
+		/// <param name="connection">The connection on which to start the transaction.</param>
+		/// <param name="cancellationToken">A cancellation token that can be used to cancel the work</param>
+		/// <returns>The started <see cref="DbTransaction" />.</returns>
+		public virtual Task<DbTransaction> BeginTransactionAsync(IsolationLevel isolationLevel, DbConnection connection, CancellationToken cancellationToken)
+		{
+			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<DbTransaction>(cancellationToken);
+			}
+			try
+			{
+				if (isolationLevel == IsolationLevel.Unspecified)
+				{
+					return connection.BeginTransactionAsync(cancellationToken);
```

Comment:
@maca88 I tried to fix this by using 'alwaysAwait' option, but I could not quite figure out the syntax.

Reply:
To fix this change the `alwaysAwait` section to:
```
alwaysAwait:
    - name: BeginTransaction
``` 
The filter `BeginTransactionAsync` could also work but the `containingTypeName` filter would need to be changed from `AsyncExtensions` to `DriverBase`.

Reply:
How does it work? I thought it identifies the method that should be always awaited.

Reply:
With `alwaysAwait` you define filters that are applied on every genereated async method to check whether to force await invocations inside the method body. So if we take the above filter as an example, every generated async method that has a sync counterpart named "BeginTransaction" will be generated by using `async/await` even if some of them could be genereated without it.